### PR TITLE
Add session usage to function perform get request

### LINKED ISSUE #6: Add session parameter to def `perform_get_request` function
Check `perform_post_request` for an example

### PR DESCRIPTION
**Pull request by MaxxxZu**
2  tests for perform_get_request added, def perform_get_request is with session yet

### CODE REVIEW

File: tests/test_parser.py [hunk 1]
```diff
@@ -21,44 +21,79 @@
...
+        session.get.assert_called_with(url=expected_url, data=None,
+                                       headers=expected_headers)
 
     def test_login_page_has_expected_title(self):
         html = '<html><title>Smarsy - Смарсі - Україна</title></html>'
         actual = validate_title(html)
         self.assertTrue(actual)
 
+
+class TestsPostPage(unittest.TestCase):
```
Comment (dkultasev) [inline tests/test_parser.py:95]: the name is wrong. we do not post pages.

File: tests/test_parser.py [hunk 2]
```diff
@@ -21,44 +21,79 @@
...
+            get=MagicMock(
+                return_value=Mock(status_code=404)
+            )
+        )
+        self.assertRaises(requests.HTTPError, perform_get_request, session,
+                          url)
+
+    def test_perform_get_request_uses_provided_data_for_get_request(
+            self):
+        expected_data = 'data'
```
Comment (dkultasev) [inline tests/test_parser.py:60]: is `data` needed for get requests?

File: tests/test_parser.py [hunk 3]
```diff
@@ -21,44 +21,79 @@
...
-    def tearDown(self):
-        pass
-
-    @patch('requests.get')
-    def test_perform_get_request_uses_provided_url_for_request(
-            self,
-            mock_request):
-        mock_request.return_value.status_code = 200
+    def test_perform_get_request_uses_provided_url_for_request(self):
+        session = Mock(
```
Comment (dkultasev) [inline tests/test_parser.py:26]: you mock the session in every single test. We might want to create new class for get requests testing and patch it as a decorator with some predefined values and if needed adjust needed properties/methods/etc in the specific test

Comment (dkultasev) [reply tests/test_parser.py:26]: As  I see that the same is done with post function tests, then let's create separate issue to fix that

Comment (dkultasev) [reply tests/test_parser.py:26]: https://github.com/dkultasev/smarsy-py/issues/8 issue is created for that

File: tests/test_parser.py (hunk 2 above)
Comment (MaxxxZu) [reply tests/test_parser.py:60]: yes, to parse childs data page we need paste data in url like
https://smarsy.ua/parent.php', params=param, headers=headers

File: tests/test_parser.py (hunk 1 above)
Comment (MaxxxZu) [reply tests/test_parser.py]: TestsPostPage renamed to TestsPostRequest

File: tests/test_parser.py (hunk 2 above)
Comment (dkultasev) [reply tests/test_parser.py:60]: if it is named differently in the original function (params) then it is better to use the same name

Comment (MaxxxZu) [reply tests/test_parser.py]: variable data in perform_get_request changed to params

**Review by dkultasev — APPROVED** [review]

**Pull request merged**